python/rmm/_lib/memory_resource.pyx

```diff
@@ -35,6 +36,42 @@ from rmm._cuda.gpu import (
 
 from rmm._lib.cuda_stream_view cimport cuda_stream_view
 
+# Transparent handle of a C++ exception
+ctypedef pair[int, string] CppExcept
+
+cdef CppExcept translate_python_except_to_cpp(err: BaseException):
+    """Translate a Python exception into a C++ exception handle
+
+    The returned exception handle can then be thrown by `throw_cpp_except()`,
+    which MUST be done without holding the GIL.
+
+    This is useful when C++ calls a Python function and needs to catch or
+    propagate exceptions.
+    """
+    if isinstance(err, MemoryError):
+        return CppExcept(0, str.encode(str(err)))
+    if isinstance(err, BaseException):
+        return CppExcept(-1, str.encode(str(err)))
```

Comment:
Let's make it clear that this is a base case, all exceptions subclass from `BaseException`.
```suggestion
    return CppExcept(-1, str.encode(str(err)))
```

Reply:
Good point!